src/components/ProgramRecordsList.jsx

```diff
@@ -0,0 +1,123 @@
+import { logError } from '@edx/frontend-platform/logging';
+import _ from 'lodash';
+import getProgramRecords from './data/service';
+
+const ProgramRecordsList = () => {
+  const renderProfile = () => {
+    const { username } = getAuthenticatedUser();
+    return (
+      <Hyperlink
+        destination={`${getConfig().LMS_BASE_URL}/u/${username}`}
+        className="mt-2 mb-4 d-inline-block muted-link pl-3 pr-3"
+      >
+        <ChevronLeft className="mb-1" />
+        <FormattedMessage
+          id="records.profile.link"
+          defaultMessage="Back to My Profile"
+          description="Link text that redirects logged-in user to their profile page"
+        />
+      </Hyperlink>
+    );
+  };
+
+  const renderCredentialsServiceIssueAlert = () => (
+    <div tabIndex="-1">
+      <Alert variant="danger">
+        <Info className="text-danger-500 mr-2 mb-1" />
+        <FormattedMessage
+          id="records.list.error"
+          defaultMessage="An error occurred attempting to retrieve your program records. Please try again later."
+          description="An error message indicating there is a problem retrieving the user's program records"
+        />
+      </Alert>
+    </div>
+  );
+
+  const renderEmpty = () => (
+    <p className="pl-3 pr-3">
+      <FormattedMessage
+        id="records.list.empty"
+        defaultMessage="No records yet. Program records are created once you have earned at least one course
+        certificate in a program."
+        description="A message indicating the user has no program records to display on the Learner Records page"
+      />
+    </p>
+  );
+
+  const [records, setRecords] = React.useState([]);
+
+  const renderRows = () => (
+    <section id="program-records-list" className="pl-3 pr-3">
+      <header>
+        <h3 className="h4">
+          <FormattedMessage
+            id="records.subheader"
+            defaultMessage="Program Records"
+            description="Subheader for the Learner Records page"
+          />
+        </h3>
+        <p>
+          <FormattedMessage
+            id="records.description"
+            defaultMessage="A program record is created once you have earned at least one course certificate
+            in a program."
+            description="Description of program records for the Learner Records page"
+          />
+        </p>
+      </header>
+    </section>
+  );
+
+  const renderList = () => {
+    const [isLoaded, setIsLoaded] = React.useState(false);
+    const [hasNoData, setHasNoData] = React.useState(false);
+
+    React.useEffect(() => {
+      getProgramRecords().then((data) => {
+        setIsLoaded(true);
+        if (_.isEmpty(data)) {
+          setHasNoData(true);
+        } else {
+          setRecords(data.enrolled_programs);
+        }
+      }).catch((error) => {
+        const errorMessage = (`Error: Could not fetch learner record data for user: ${error.message}`);
+        logError(errorMessage);
+      });
+    }, []);
```

Comment:
I’m used to seeing these things at the top of the React functional component (so right after line 12 here). Can you move them up? I’m not sure how putting them within a function that’s called in the return (`renderList` is called in the `return` below) might affect things. Maybe not at all! But unless there’s a reason for putting them them within `renderList` specifically, I’d move them up so that they are where developers are used to looking for them.